cirq/circuits/circuit_dag.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2018 The ops Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Generic, Iterator, Type, TypeVar
+
+import networkx
+
+from cirq import ops, devices, abc
+from cirq.circuits import circuit
+
+
+T = TypeVar('T')
+
+class Unique(Generic[T]):
```

Comment:
Add a docstring explaining why this is needed.

Reply:
Done.